### ISSUE TITLE
Google one-billion experiments

### ISSUE THREAD

**Issue by jiahuigeng** [kimiyoung/transformer-xl#53]
with config: python train.py --cuda --data ../data/one-billion-words/ --dataset lm1b --adaptive --n_layer 18 --d_model 1024 --div_val 4 --n_head 8 --d_head 128 --d_inner 4096 --dropout 0.0 --dropatt 0.0 --optim adam --log-interval 5 --eval-interval 20 --warmup_step 20000 --max_step 500000 --lr 0.00025 --tgt_len 32 --mem_len 32 --eval_tgt_len 32 --batch_size 240 --batch_chunk 8 --work_dir exps, so run on only one GPU, do you think possible to achieve similar results as in the paper?

**Comment by kimiyoung** [comment]
I think it's almost impossible to get competitive results with one gpu, unfortunately.

**Comment by jiahuigeng** [comment]
> I think it's almost impossible to get competitive results with one gpu, unfortunately.

What is your recommended batch size and number of GPUs for this experiments?
Sorry, I didn't get such information from the paper

**Comment by kimiyoung** [comment]
https://github.com/kimiyoung/transformer-xl/tree/master/tf#3-resources-needed-for-sota-model-training

These are resource requirements for our sota settings. We also provide base model training scripts that require less resources. You might play with it to find one that fits your hardware.

**Comment by sanwei111** [comment]
excuse me，can you please explain the Class “class RelPartialLearnableMultiHeadAttn(RelMultiHeadAttn):”.I think the cLASS CAN'T MATCH THE FORMULAS IN THE PAPER SO MUCH!
